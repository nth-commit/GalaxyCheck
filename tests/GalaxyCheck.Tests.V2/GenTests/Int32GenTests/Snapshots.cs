using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
