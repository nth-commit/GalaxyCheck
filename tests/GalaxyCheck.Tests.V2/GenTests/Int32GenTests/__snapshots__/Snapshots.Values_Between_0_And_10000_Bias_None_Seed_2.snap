[
  7711,
  7129,
  9506,
  7430,
  197,
  7282,
  2618,
  9587,
  8918,
  2374,
  6301,
  6497,
  1737,
  6247,
  386,
  4837,
  7372,
  6529,
  761,
  3630,
  1614,
  5894,
  3326,
  6881,
  4347,
  4769,
  3955,
  6892,
  3128,
  4024,
  6106,
  9141,
  9645,
  780,
  2554,
  5210,
  6416,
  5811,
  2226,
  9161,
  1420,
  6801,
  6816,
  7478,
  9415,
  4926,
  1388,
  2290,
  3609,
  4128,
  8853,
  1382,
  5534,
  5010,
  8444,
  4603,
  9689,
  6478,
  8282,
  8944,
  2441,
  5001,
  8384,
  3249,
  9266,
  2923,
  3801,
  7940,
  9089,
  2741,
  9830,
  2648,
  9179,
  4048,
  3582,
  3312,
  7755,
  6228,
  1894,
  4742,
  5480,
  1649,
  9154,
  4437,
  456,
  9890,
  8049,
  7867,
  3005,
  4636,
  598,
  4170,
  9243,
  7827,
  3255,
  706,
  1583,
  1194,
  6135,
  4373
]
